### PR TITLE
Dionae changes

### PR DESCRIPTION
**Pull request by Yonnimer**
PR is still work in progress. 

- Increases nymph health to 50

- Changes Dionae health to 240(currently from 300)

- When a dionae splits their health is now divided amongst their nymphs.  Nymphs will always have at least 20% of their health when split.

### CODE REVIEW

**Comment by Yonnimer** [comment]
!wip

File: code/modules/mob/living/carbon/human/species/station/diona/diona.dm [hunk 1]
```diff
@@ -160,7 +161,7 @@
 /datum/species/diona/handle_death(var/mob/living/carbon/human/H, var/gibbed = 0)
 	if (!gibbed)
 		// This proc sleeps. Async it.
-		INVOKE_ASYNC(H, /mob/living/carbon/human/proc/diona_split_into_nymphs, TRUE)
+		INVOKE_ASYNC(H, /mob/living/carbon/human/proc/diona_split_into_nymphs, total_health * 0.20)
```
Comment (skull132) [inline code/modules/mob/living/carbon/human/species/station/diona/diona.dm:164]: that's uh. Not how this works. The value is expected to be false-y or truth-y. The result of computation that's other than 0 is truthy.

Comment (Yonnimer) [reply code/modules/mob/living/carbon/human/species/station/diona/diona.dm]: Removed this part entirely given a check already handled it.

**Comment by ParadoxSpace** [comment]
For what reason, in detail, is a health buff required.

File: code/modules/mob/living/carbon/human/human_damage.dm [hunk 2]
```diff
@@ -1,6 +1,6 @@
 //Updates the mob's health from organs and mob damage variables
 /mob/living/carbon/human/updatehealth()
-	if(is_diona())
+	if(IS_DIONA())
```
Comment (Arrow768) [inline code/modules/mob/living/carbon/human/human_damage.dm:3]: Why did you change that?
IS_DIONA is (variable) define used in chemistry

File: code/modules/mob/living/carbon/human/life.dm [hunk 3]
```diff
@@ -78,6 +78,8 @@
 		handle_stamina()
 
 		if (is_diona())
+			if(health < -40) //240+ damage
+				diona_split_into_nymphs()
```
Comment (Arrow768) [inline code/modules/mob/living/carbon/human/life.dm:82]: That wont work. As the diona already splits at 0 health due to the death -> It will never be able to reach -40

Comment (Yonnimer) [reply code/modules/mob/living/carbon/human/life.dm]: Not 100% what I was doing. Dionae health is back to 300, and they split at 60 health.

File: code/modules/mob/living/carbon/human/human_damage.dm (hunk 2 above)
Comment (Yonnimer) [reply code/modules/mob/living/carbon/human/human_damage.dm]: Forgot to change this back when I was testing something. Fixed now.

File: html/changelogs/Dionahealth.yml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+delete-after: True
+
+# Any changes you've made.  See valid prefix list above.
+# INDENT WITH TWO SPACES.  NOT TABS.  SPACES.
+# SCREW THIS UP AND IT WON'T WORK.
+# Also, all entries are changed into a single [] after a master changelog generation. Just remove the brackets when you add new entries.
+# Please surround your changes in  double quotes ("), as certain characters otherwise screws up compiling. The quotes will not show up in the changelog.
+changes: 
+  - rsctweak: "Changes Dionae health to 300, from 200"
+  - rsctweak: "Changes nymph health to 50, from 33.3"
```
Comment (Arrow768) [inline html/changelogs/Dionahealth.yml:43]: rsctweak does not exist

Comment (Yonnimer) [reply html/changelogs/Dionahealth.yml]: Fixed

**Comment by Yonnimer** [comment]
!review

**Comment by ParadoxSpace** [comment]
For what reason, in detail, is a health buff required.

**Comment by Yonnimer** [comment]
@ParadoxSpace Arrow wants Dionae health = Nymph health * amount of nymphs that make up the Dionae. So for 50 nymph health, it would need 300 Dionae health. Nymphs need more health as 33.3 is too damn low for a species meant to be resilient.  

You also need to look at more than the buff. This pr makes Dionae split at 240 health, making their actual health 240, not 300, as they'll never get past 240 damage. With nymphs also being damaged based on the Dionae's damage, it turns their combined pool into 300 health, no matter what. Even with the nymph and Dionae health this combined pool is lower than before, which prior it was about 399.8 health

File: code/modules/mob/living/carbon/human/life.dm [hunk 5]
```diff
@@ -78,6 +78,8 @@
 		handle_stamina()
 
 		if (is_diona())
+			if(health < 60)
+				diona_split_into_nymphs()
```
Comment (skull132) [inline code/modules/mob/living/carbon/human/life.dm:82]: Not sure if I like this, as I said. It's relatively needless, since we already split properly at death. And will add complexity when it can be avoided.

Comment (Yonnimer) [reply code/modules/mob/living/carbon/human/life.dm:82]: What would be a good way to do it then? The point of this is to make sure nymphs no matter what have at least 20% of their health, so 60 health remaining / the 6 nymphs is 10 health(20% of their health the pr)

**Comment by Yonnimer** [comment]
The two above changes are WIP. Need to discuss it with skull.

**Comment by Yonnimer** [comment]
!wip

File: code/modules/mob/living/carbon/human/diona_gestalt.dm [hunk 6]
```diff
@@ -301,11 +301,10 @@
 	else
 		nymph.key = key
 
-/mob/living/carbon/human/proc/diona_split_into_nymphs(var/death_split = FALSE)
+/mob/living/carbon/human/proc/diona_split_into_nymphs()
 	var/turf/T = get_turf(src)
 	var/mob/living/carbon/alien/diona/bestNymph = null
-	var/bestHealth = 0
-
+	var/gestalt_health = 300 -  getFireLoss() - getBruteLoss() - getToxLoss()
```
[the remark this answers is not in the document]
Comment (Yonnimer) [reply code/modules/mob/living/carbon/human/diona_gestalt.dm]: Done

File: code/modules/mob/living/carbon/human/species/station/diona/diona.dm [hunk 7]
```diff
@@ -160,7 +161,7 @@
 /datum/species/diona/handle_death(var/mob/living/carbon/human/H, var/gibbed = 0)
 	if (!gibbed)
 		// This proc sleeps. Async it.
-		INVOKE_ASYNC(H, /mob/living/carbon/human/proc/diona_split_into_nymphs, TRUE)
+		INVOKE_ASYNC(H, /mob/living/carbon/human/proc/diona_split_into_nymphs)
```
Comment (NonQueueingMatt) [inline code/modules/mob/living/carbon/human/species/station/diona/diona.dm:164]: ??? What the hell is going on here? Can you walk me through what this code does, because I seriously don't understand why you're lowering diona health but then later doing ` 	var/gestalt_health = (300 -  getFireLoss() - getBruteLoss() - getToxLoss()) / 6 `.

**Comment by Yonnimer** [comment]
I'm trying to make it so Dionae health = Nymph health * amount of the nymphs  in the dionae. Dionae would have 300 health, but to give them a buffer so the nymphs don't instantly die when deathsplitting the Dionae's health is set to 240, the last 60 from the 300 is divided between the nymphs so they get 10 health no matter what. Originally they were just going to split once they reach 240 damage and have 300 health but this caused issues, so it was changed to this. The 300 health from `var/gestalt_health = (300 - getFireLoss() - getBruteLoss() - getToxLoss()) / 6.` is the Dionae's total health, plus it helps simplify the math.

File: code/modules/mob/living/carbon/human/species/station/diona/diona.dm (hunk 7 above)
Comment (Yonnimer) [reply code/modules/mob/living/carbon/human/species/station/diona/diona.dm:164]: I'm trying to make it so Dionae health = Nymph health * amount of the nymphs in the dionae. Dionae would have 300 health, but to give them a buffer so the nymphs don't instantly die when deathsplitting the Dionae's health is set to 240, the last 60 from the 300 is divided between the nymphs so they get 10 health no matter what. Originally they were just going to split once they reach 240 damage and have 300 health but this caused issues, so it was changed to this. The 300 health from var/gestalt_health = (300 - getFireLoss() - getBruteLoss() - getToxLoss()) / 6. is the Dionae's total health, plus it helps simplify the math.

**Comment by Yonnimer** [comment]
I changed it back to 240 as Dionae should be splitting only once they reach 0 health now. So it is a health buff again.

**Comment by ParadoxSpace** [comment]
Their regeneration will probably need looking at.

**Comment by Yonnimer** [comment]
Going back to 140 total health since dionae split at -100 again as of https://github.com/Aurorastation/Aurora.3/pull/9254
